### PR TITLE
sql: add a note about `FULLTEXT`

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
```
mysql> CREATE TABLE t1 (id INT PRIMARY KEY, v VARCHAR(255), INDEX (v) USING BTREE);
Query OK, 0 rows affected (0.01 sec)

mysql> CREATE TABLE t2 (id INT PRIMARY KEY, v VARCHAR(255), INDEX (v) USING HASH);
Query OK, 0 rows affected (0.01 sec)

mysql> SHOW CREATE TABLE t1\G
*************************** 1. row ***************************
       Table: t1
Create Table: CREATE TABLE `t1` (
  `id` int(11) NOT NULL,
  `v` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */,
  KEY `v` (`v`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
1 row in set (0.00 sec)

mysql> SHOW CREATE TABLE t2\G
*************************** 1. row ***************************
       Table: t2
Create Table: CREATE TABLE `t2` (
  `id` int(11) NOT NULL,
  `v` varchar(255) DEFAULT NULL,
  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */,
  KEY `v` (`v`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
1 row in set (0.00 sec)

mysql> SHOW INDEX FROM t1;
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
| Table | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment | Visible | Expression | Clustered |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
| t1    |          0 | PRIMARY  |            1 | id          | A         |           0 |     NULL | NULL   |      | BTREE      |         |               | YES     | NULL       | YES       |
| t1    |          1 | v        |            1 | v           | A         |           0 |     NULL | NULL   | YES  | BTREE      |         |               | YES     | NULL       | NO        |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
2 rows in set (0.00 sec)

mysql> SHOW INDEX FROM t2;
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
| Table | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment | Visible | Expression | Clustered |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
| t2    |          0 | PRIMARY  |            1 | id          | A         |           0 |     NULL | NULL   |      | BTREE      |         |               | YES     | NULL       | YES       |
| t2    |          1 | v        |            1 | v           | A         |           0 |     NULL | NULL   | YES  | HASH       |         |               | YES     | NULL       | NO        |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
2 rows in set (0.00 sec)

```

So we support `FULLTEXT` and `HASH` syntax wise. Where `FULLTEXT` results in a warning, this isn't the case for `HASH`.  Note that the index type for hash persists in the output of `SHOW INDEX`, but not in the output of `SHOW CREATE TABLE`.

**Comment by dveeden** [comment]
Note that these pages should have a note about `FULLTEXT`:
- `sql-statements/sql-statement-add-index.md`
- `sql-statements/sql-statement-create-index.md`
- `sql-statements/sql-statement-create-table.md`

Note that the "CREATE INDEX" page already has a complete note about this: "TiDB supports parsing the `FULLTEXT` and `SPATIAL` syntax but does not support using the `FULLTEXT`, `HASH`, and `SPATIAL` indexes."

Note that `mysql-compatibility.md` has a note about HASH being supported, but ignored.

**Comment by dveeden** [comment]
@kennytm looks like "USING" for index type is missing from the EBNF for "CREATE TABLE"?

**Comment by dveeden** [comment]
```
mysql> CREATE TABLE t3 (id INT PRIMARY KEY, v VARCHAR(255), INDEX (v) USING RTREE);
Query OK, 0 rows affected (0.01 sec)

mysql> SHOW INDEX FROM t3;
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
| Table | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment | Visible | Expression | Clustered |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
| t3    |          0 | PRIMARY  |            1 | id          | A         |           0 |     NULL | NULL   |      | BTREE      |         |               | YES     | NULL       | YES       |
| t3    |          1 | v        |            1 | v           | A         |           0 |     NULL | NULL   | YES  | RTREE      |         |               | YES     | NULL       | NO        |
+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+-----------+
2 rows in set (0.00 sec)
```
Note that `RTREE` is also accepted as index type (and ignored).

File: sql-statements/sql-statement-add-index.md [hunk 1]
```diff
@@ -85,7 +85,8 @@ mysql> EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 
 ## MySQL compatibility
 
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* `HASH` and `SPATIAL` indexes are not supported.
```
Comment (dveeden) [inline sql-statements/sql-statement-add-index.md:88]:
```suggestion
* Index types like `HASH`, `BTREE` and `RTREE` are accepted in syntax but ignore for compatibility reasons.
* `SPATIAL` indexes are not supported.
```

File: sql-statements/sql-statement-add-index.md [hunk 2]
```diff
@@ -85,7 +85,8 @@ mysql> EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
...
+* `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.
```
Comment (dveeden) [inline sql-statements/sql-statement-add-index.md:89]:
```suggestion
* `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.
* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.
```

**Comment by dveeden** [comment]
Not that ``TiDB supports parsing the `FULLTEXT` and `SPATIAL` syntax`` from `sql-statements/sql-statement-create-index.md` is incorrect as far as I know. I don't think we support `...SPATIAL INDEX...` syntax.

**Comment by dveeden** [comment]
Maybe we should also add a note about the (meaningless) `Index_type` in the output to this page? https://docs.pingcap.com/tidb/stable/sql-statement-show-indexes

**Comment by hfxsd** [comment]
> Maybe we should also add a note about the (meaningless) `Index_type` in the output to this page? [docs.pingcap.com/tidb/stable/sql-statement-show-indexes](https://docs.pingcap.com/tidb/stable/sql-statement-show-indexes)

OK, updated.

File: sql-statements/sql-statement-add-index.md [hunk 3]
```diff
@@ -85,7 +85,10 @@ mysql> EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 
 ## MySQL compatibility
 
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
+* `SPATIAL` indexes are not supported.
+* `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.
+* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.
```
Comment (dveeden) [inline sql-statements/sql-statement-add-index.md:91]: I think lines 90 and 91 are saying the same thing now

File: sql-statements/sql-statement-create-table.md [hunk 4]
```diff
@@ -238,8 +238,10 @@ mysql> DESC t1;
 
 ## MySQL compatibility
 
-* All of the data types except spatial types are supported.
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* All of the data types except `SPATIAL` types are supported.
+* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
+* `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.
```
Comment (dveeden) [inline sql-statements/sql-statement-create-table.md:243]: 243 and 244 are now saying the same. I like the second line more than the first.

**Review by dveeden — APPROVED** [review]

File: sql-statements/sql-statement-create-table.md (hunk 4 above)
Comment (hfxsd) [inline sql-statements/sql-statement-create-table.md:243]:
```suggestion
```

File: sql-statements/sql-statement-add-index.md [hunk 5]
```diff
@@ -85,7 +85,10 @@ mysql> EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 
 ## MySQL compatibility
 
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
+* `SPATIAL` indexes are not supported.
+* `FULLTEXT` is only for syntax compatibility with MySQL and is not supported.
```
Comment (hfxsd) [inline sql-statements/sql-statement-add-index.md:90]:
```suggestion
```

File: sql-statements/sql-statement-add-index.md (hunk 3 above)
Comment (hfxsd) [reply sql-statements/sql-statement-add-index.md:90]: Ok deleted one

File: sql-statements/sql-statement-create-table.md (hunk 4 above)
Comment (hfxsd) [reply sql-statements/sql-statement-create-table.md]: ok deleted the first one.

**Comment by dveeden** [comment]
I think the IndexOption of the create table statement page should be changed to:
```ebnf+diagram
IndexOption ::=
    'COMMENT' String
|   ( 'VISIBLE' | 'INVISIBLE' )
|   ('USING' | 'TYPE') ('BTREE' | 'RTREE' | 'HASH')
```

**Comment by dveeden** [comment]
Looks like these pages reference `IndexInvisible` but that's never defined. cc @kennytm

File: sql-statements/sql-statement-create-table.md [hunk 6]
```diff
@@ -238,8 +238,9 @@ mysql> DESC t1;
 
 ## MySQL compatibility
 
-* All of the data types except spatial types are supported.
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* All of the data types except `SPATIAL` types are supported.
```
Comment (kennytm) [inline sql-statements/sql-statement-create-table.md:241]: Spatial type means https://dev.mysql.com/doc/refman/8.4/en/spatial-type-overview.html i.e. `GEOMETRY`, `POLYGON`, `MULTIPOINT` etc. There isn't a single type called `SPATIAL`, and this item is not talking about spatial index either.

**Comment by kennytm** [comment]
@dveeden  If any page references `IndexInvisible` the node should be replaced with `("VISIBLE" | "INVISIBLE")`.

If you want a hyperlink it should go to https://docs.pingcap.com/tidb/stable/sql-statement-alter-index.

File: sql-statements/sql-statement-create-table.md (hunk 6 above)
Comment (hfxsd) [inline sql-statements/sql-statement-create-table.md:241]:
```suggestion
* All of the data types except spatial types are supported.
```

File: sql-statements/sql-statement-add-index.md [hunk 7]
```diff
@@ -85,7 +85,9 @@ mysql> EXPLAIN SELECT * FROM t1 WHERE c1 = 3;
 
 ## MySQL compatibility
 
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
+* `SPATIAL` indexes are not supported.
+* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.
```
Comment (lilin90) [inline sql-statements/sql-statement-add-index.md:90]:
```suggestion
* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax for compatibility with MySQL, but ignores them.
* `SPATIAL` indexes are not supported.
* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT` indexes.
```

File: sql-statements/sql-statement-create-table.md [hunk 8]
```diff
@@ -239,7 +239,8 @@ mysql> DESC t1;
 ## MySQL compatibility
 
 * All of the data types except spatial types are supported.
-* `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
+* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
```
Comment (lilin90) [inline sql-statements/sql-statement-create-table.md:242]:
```suggestion
* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax for compatibility with MySQL, but ignores them.
```

File: sql-statements/sql-statement-show-indexes.md [hunk 9]
```diff
@@ -53,6 +53,8 @@ mysql> SHOW KEYS FROM t1;
 2 rows in set (0.00 sec)
 ```
 
+Note that TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
```
Comment (lilin90) [inline sql-statements/sql-statement-show-indexes.md:56]:
```suggestion
Note that TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax for compatibility with MySQL, but ignores them.
```

File: sql-statements/sql-statement-create-index.md [hunk 10]
```diff
@@ -349,7 +349,8 @@ The system variables associated with the `CREATE INDEX` statement are `tidb_ddl_
 
 ## MySQL compatibility
 
-* TiDB supports parsing the `FULLTEXT` and `SPATIAL` syntax but does not support using the `FULLTEXT`, `HASH`, and `SPATIAL` indexes.
+* TiDB supports parsing the `FULLTEXT` syntax but does not support using the `FULLTEXT`, `HASH`, and `SPATIAL` indexes.
+* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax but ignores them for compatibility reasons.
```
Comment (lilin90) [inline sql-statements/sql-statement-create-index.md:353]:
```suggestion
* TiDB accepts index types such as `HASH`, `BTREE` and `RTREE` in syntax for compatibility with MySQL, but ignores them.
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**